addons/api/addon/serializers/auth-method.js

```diff
@@ -17,18 +17,35 @@ export default class AuthMethodSerializer extends ApplicationSerializer {
    */
   serialize(snapshot) {
     switch (snapshot.record.type) {
-      case TYPE_AUTH_METHOD_OIDC:
-        return this.serializeOIDC(...arguments);
+      case TYPE_AUTH_METHOD_PASSWORD:
+        return this.serializePassword(...arguments);
       default:
         return this.serializeDefault(...arguments);
     }
   }
 
+  serializeAttribute(snapshot, json, key, attribute) {
+    super.serializeAttribute(...arguments);
+    const { type } = snapshot.record;
+    const { options } = attribute;
+
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excludedOptional user-set description for identification purposes
+    // from serialization.
+    if (options?.for && options.for !== type) {
```

Comment:
I think you might need to be able to handle the `for` as an array as well? This might be worth pulling out into the application serializer at this point.